src/components/TodoInfo.tsx

```diff
@@ -0,0 +1,164 @@
+import { useEffect, useRef, useState } from 'react';
+import cn from 'classnames';
+import { UpdateTodo, deleteTodos } from '../api/todos';
+import { showError } from '../helpers/helpers';
+import { Todo } from '../types/Todo';
+
+interface Props {
+  setError: React.Dispatch<React.SetStateAction<string>>,
+  setTodos: React.Dispatch<React.SetStateAction<Todo[]>>,
+  loader: number[],
+  setLoader:React.Dispatch<React.SetStateAction<number[]>>
+  todo: Todo,
+}
+
+export const TodoInfo:React.FC<Props> = ({
+  setError,
+  setTodos,
+  loader,
+  setLoader,
+  todo,
+}) => {
+  const [onEdit, setOnEdit] = useState<null | number>(null);
```

Comment:
you can just use boolean flag here
```suggestion
  const [isEditing, setIsEditing] = useState(false);
```